apps/workshop/homarr_provider.go

```
@@ -0,0 +1,475 @@
+	upstreamRequest.Header.Set("Content-Type", "application/json")
+	upstreamRequest.Header.Set("Accept", "text/event-stream, application/json")
+	upstreamRequest.Header.Set("HTTP-Referer", "https://homarr.dev")
+	upstreamRequest.Header.Set("X-Title", "Homarr Provider")
+	upstreamRequest.Header.Set("X-OpenRouter-Metadata", "enabled")
+
+	upstreamResponse, err := provider.httpClient.Do(upstreamRequest)
+	if err != nil {
+		provider.finishActivity(event.App, privateRequest, publicActivity, "failed", requestUsage{}, startedAt)
+		return event.JSON(http.StatusBadGateway, openAIError("The model endpoint could not be reached."))
+	}
+	defer upstreamResponse.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `upstreamResponse.Body.Close` is not checked

(errcheck)